spec/controllers/publications_controller_spec.rb

```diff
@@ -0,0 +1,214 @@
+require 'rails_helper'
+
+RSpec.describe PublicationsController do
+  #Factories
+  let(:publication) { create(:publication) }
+  let(:user) { create(:user) }
+  #Methods
+  def get_index
+    get :index
+  end
+
+  def get_new
+    get :new
+  end
+
+  def post_create
+    post :create, params: {publication: attributes_for(:publication)}
+  end
+
+  def get_show
+    get :show, params: {id: publication}
+  end
+
+  def get_edit
+    get :edit, params: {id: publication}
+  end
+
+  def put_update
+    put :update, params: {id: publication, publication: attributes_for(:article, title: "Anna")}
+  end
+
+  def permission_granted
```

Comment:
Rename this make_publication_with_member. change all instances of it being called.